### PR TITLE
Add template for MM Program certificate

### LINKED ISSUE #3519: Add template for program certificates
As a learner, I'd like to be able to view an HTML certificate when I complete a FA program (non-FA program certificates will be available from edx.org). I'd like also to be able to print the certificate from my browser. 

screen mock-up at https://invis.io/NB7VM1GSG#/239889459_Program_Complete2

### questions
- [ ]  Same URL as course certificates?

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #3519

#### What's this PR do?
Add cms model for `ProgramCertificateSignatories`, add a view for program certificate.

#### How should this be manually tested?
create a program certificate 
```from django.contrib.auth.models import User
user = User.objects.filter(username='username')[0]

from courses.models import *
from grades.models import *
from grades.factories import *

program = Program.objects.filter(title='Digital Learning')[0]
for c in program.course_set.all():

    course_run = c.courserun_set.all().first()

    ### Create grade, complete status, and certificate
    final_grade = FinalGradeFactory.create(user=user, course_run=course_run, passed=True, status='complete', grade=0.8)
    CourseRunGradingStatus.objects.create(course_run=course_run, status='complete')
    cert = MicromastersCourseCertificate.objects.create(final_grade=final_grade)
```
Once all the courses in the program have a certificate, the program certificate should get created automatically.
In the cms admin create `ProgramCertificateSignatories` for the program. And then you should be able to view the certificate.

### CODE REVIEW

File: certificates/views.py [hunk 1]
```diff
@@ -52,3 +66,37 @@ def get_context_data(self, **kwargs):
...
         return context
+
+
+class ProgramCertificateView(CertificateView):
+    """
+    Views for program certificates
+    """
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
```
Comment (noisecapella) [inline certificates/views.py]: Can you write tests covering these cases? You can access the context of the view and assert its values using the test client

**Comment by annagav** [comment]
@noisecapella, ready for review!

File: ui/templates/certificate.html [hunk 2]
```diff
@@ -19,12 +19,24 @@
...
+          <p>
+            has successfully completed all the courses and received passing grades for a MicroMasters credential
+            inpassed all the requirements of the
+          </p>
+          <p class="course-name">{{ program_title }}</p>
+          <p>
+            a program offered by MITx, in collaboration with edX.
+          </p>
+        {% endif %}
+
```
Comment (noisecapella) [inline ui/templates/certificate.html]: The link below (the 'valid certificate id' link) is always to a course certificate

File: ui/templates/certificate.html [hunk 3]
```diff
@@ -19,12 +19,24 @@
...
       </div>
 
       <div class="certificate-stamps row">
```
Comment (noisecapella) [inline ui/templates/certificate.html]: There's a typo below: `inpassed`, and the `the {{ program_title }} a program offered by MITx` seems like strange wording. @roberthouse54 any input here?

File: ui/templates/certificate.html (hunk 2 above)
Comment (noisecapella) [reply ui/templates/certificate.html]: Please also add a test for this, for the course and program certificate links

File: ui/templates/certificate.html [hunk 4]
```diff
@@ -19,12 +19,24 @@
...
+            the Massachusetts Institute of Technology in collaboration with edX.
+          </p>
+        {% else %}
+          <p>
+            has successfully completed all the courses and received passing grades for a MicroMasters credential
+            inpassed all the requirements of the
+          </p>
+          <p class="course-name">{{ program_title }}</p>
+          <p>
+            a program offered by MITx, in collaboration with edX.
```
Comment (noisecapella) [inline ui/templates/certificate.html]: The wording here `MITx` should be changed to be consistent with `Massachusetts Institute of Technology`, can you ask @roberthouse54 about this too?

Comment (noisecapella) [reply ui/templates/certificate.html]: Also, what's the difference between this and `signatory.organization`?

File: ui/templates/certificate.html (hunk 3 above)
Comment (roberthouse54) [reply ui/templates/certificate.html]: Yes, that was my mistake... typos in the mock. That should read:

"This is to certify that [name] has successfully completed all the courses and received passing grades for a MicroMasters credential in [program_title] a program offered by MITx, in collaboration with edX."

**Comment by noisecapella** [comment]
screenshot from 2017-09-21 10-55-54

**Comment by annagav** [comment]
@noisecapella I addressed all comments

**Pull request merged**